dds/DCPS/transport/framework/TransportRegistry.cpp

```diff
@@ -579,10 +580,87 @@ TransportRegistry::bind_config(const TransportConfig_rch& cfg,
+
+  // if domain is in a domain range and config is a transport template,
+  // get the correct config.
+  if (TheServiceParticipant->belongs_to_domain_range(ei->get_domain_id())) {
+    bool found = false;
+    for (OPENDDS_VECTOR(TransportTemplate)::const_iterator i = transport_templates_.begin(); i != transport_templates_.end(); ++i) {
+      if (cfg->name() == i->config_name) {
+        found = true;
+        break;
+      }
+    }
+
+    if (found)
+    {
+      ACE_TString cfg_name = ACE_TEXT_CHAR_TO_TCHAR(cfg->name().c_str());
+      // create if not already created
+      int ret = create_transport_template_instance(ei->get_domain_id(), cfg_name);
+
+      if (ret == 0) {
+        // get guid and create unique name for transport instance
+        GUID_t guid = ei->get_id();
+        if (guid == GUID_UNKNOWN) {
+          ACE_ERROR((LM_ERROR,
+                     ACE_TEXT("(%P|%t) TransportRegistry::bind_config: ")
+                     ACE_TEXT("GUID_UNKNOWN. Can not bind entity to a domain template instance.\n")));
+          throw Transport::UnableToCreate();
+        }
+        OPENDDS_STRING transport_inst_name = GuidConverter(guid).uniqueId();
+        ACE_TString transport_config_name = cfg_name;
+
+        bool success = create_new_transport_instance_for_participant(ei->get_domain_id(), transport_config_name, transport_inst_name);
+
+        if (success) {
+          // update config
+          TransportConfig_rch new_cfg = get_config(ACE_TEXT_ALWAYS_CHAR(transport_config_name.c_str()));
+          update_config_template_instance_info(new_cfg->name(), transport_inst_name);
+          ei->transport_config(new_cfg);
+          return;
+        } else {
+          ACE_ERROR((LM_ERROR,
+            ACE_TEXT("(%P|%t) TransportRegistry::bind_config: ")
+            ACE_TEXT("Failed to create new transport template instance.\n")));
+          throw Transport::UnableToCreate();
+        }
+      }
+    }
+  }
+
   ei->transport_config(cfg);
 }
 
 
+void
+TransportRegistry::remove_transport_template_instance(const OPENDDS_STRING& config_name)
+{
+  ConfigTemplateToInstanceMap::const_iterator i = config_template_to_instance_map.find(config_name);
+  if (i == config_template_to_instance_map.end()) {
+    if (DCPS_debug_level > 4) {
+      ACE_DEBUG((LM_DEBUG,
+               ACE_TEXT("(%P|%t) TransportRegistry::remove_transport_template_instance: ")
+               ACE_TEXT("%C is not a transport template instance.\n"),
+               config_name.c_str()));
+      return;
+    }
+  }
+
+  OPENDDS_STRING inst_name = i->second;
+  remove_config(config_name);
+  remove_inst(inst_name);
+
+  if (DCPS_debug_level > 0) {
+    ACE_DEBUG((LM_DEBUG,
+               ACE_TEXT("(%P|%t) DomainParticipantFactoryImpl::delete_participant ")
+               ACE_TEXT("deleted TransportRegistry's dynamically created config %C and instance %C\n"),
+               config_name.c_str(), inst_name.c_str()));
+  }
+
+  config_template_to_instance_map.erase(i);
```

Comment:
This line has an error on one of the CI builds - change the type of `i` to the regular iterator (not const_iterator) to get it to work with all C++ standard versions.